model.py

```diff
@@ -75,6 +76,42 @@ def forward(self,
                 ):
         X = self.backbone(X)
         X_proj = self.projector(X)
+        X_proj = X_proj.detach()
```

Comment:
For some reason having the `.detach()` here works, which was the main error stopping learning;